metrics/config.go

```diff
@@ -327,6 +328,24 @@ import (
 )`, DomainEnv, DomainEnv))
 }
 
+// PrometheusPort returns the TCP port number configured via the environment
+// for the Prometheus metrics exporter if it's set, a default value otherwise.
+// No validation is performed on the port value.
+func PrometheusPort() int {
+	ppStr := os.Getenv(prometheusPortEnvName)
+	if ppStr == "" {
+		return defaultPrometheusPort
+	}
+
+	pp, err := strconv.Atoi(ppStr)
```

Comment:
You probably want more validation
1. strconv.ParseUint(x, 10, 16) — would catch all but 0.
2. 0 is a valid port in unix, but you won't be able to use generally, since system will assign a random port — so probably want to exclude that as well.

Reply:
My intention was really just to read an int here. Not saying your idea is not good, but the validation happens in the caller right now and I didn't want to make this PR a bigger refactoring PR.

Would it be better if I just make this function private?

Reply:
yeah, I don't see it being called elsewhere.
But you know changing strconv.Atoi to strconv.ParseUint is not a bigger refactoring :)
We can skip 0 check for existing code to take care of.

Reply:
Alright, let me revisit this. Good suggestion.

Reply:
+1 to private func.

Reply:
Done
https://github.com/knative/pkg/blob/0ec08e2b0f1d605c10ca5eba3fed4523e12dca35/metrics/config.go#L338-L355